src/measure_theory/lp_space.lean

```diff
@@ -1218,4 +1217,390 @@ by { rw eq, exact continuous_pos_part.comp continuous_neg }
+    ≤ (∑' i, B i) ^ p) :
+  (∫⁻ a, (∑' i, nnnorm (f (i + 1) a - f i a) : ℝ≥0∞)^p ∂μ) ^ (1/p) ≤ ∑' i, B i :=
+begin
+  have hp_pos : 0 < p := zero_lt_one.trans_le hp1,
+  suffices h_pow : ∫⁻ a, (∑' i, nnnorm (f (i + 1) a - f i a) : ℝ≥0∞)^p ∂μ ≤ (∑' i, B i) ^ p,
+    by rwa [←@ennreal.le_rpow_one_div_iff _ _ (1/p) (by simp [hp_pos]), one_div_one_div],
+  have h_tsum_1 : ∀ g : ℕ → ℝ≥0∞,
+      ∑' i, g i = filter.at_top.liminf (λ n, ∑ i in finset.range (n + 1), g i),
+    by { intro g, rw [ennreal.tsum_eq_liminf_sum_nat, ← filter.liminf_nat_add _ 1], },
+  simp_rw h_tsum_1 _,
+  rw ← h_tsum_1,
+  have h_liminf_pow : ∫⁻ a, filter.at_top.liminf (λ n, ∑ i in finset.range (n + 1),
+      (nnnorm (f (i + 1) a - f i a)))^p ∂μ
+    = ∫⁻ a, filter.at_top.liminf (λ n, (∑ i in finset.range (n + 1),
+      (nnnorm (f (i + 1) a - f i a)))^p) ∂μ,
+  { refine lintegral_congr (λ x, _),
+    have h_rpow_mono := ennreal.rpow_left_strict_mono_of_pos (zero_lt_one.trans_le hp1),
+    have h_rpow_surj := (ennreal.rpow_left_bijective hp_pos.ne.symm).2,
+    refine (h_rpow_mono.order_iso_of_surjective _ h_rpow_surj).liminf_apply _ _ _ _,
+    all_goals { filter.is_bounded_default }, },
+  rw h_liminf_pow,
+  refine (lintegral_liminf_le' _).trans _,
+  { exact λ n, ae_measurable.ennreal_rpow_const (finset.ae_measurable_sum (finset.range (n+1))
+      (λ i, ((hf (i+1)).sub (hf i)).nnnorm.ennreal_coe)), },
+  { exact filter.liminf_le_of_frequently_le' (filter.frequently_of_forall h), },
+end
+
+private lemma tsum_nnnorm_sub_ae_lt_top
+  {f : ℕ → α → E} (hf : ∀ n, ae_measurable (f n) μ) {p : ℝ} (hp1 : 1 ≤ p) {B : ℕ → ℝ≥0∞}
+  (hB : ∑' i, B i < ∞)
+  (h : (∫⁻ a, (∑' i, nnnorm (f (i + 1) a - f i a) : ℝ≥0∞)^p ∂μ) ^ (1/p) ≤ ∑' i, B i) :
+  ∀ᵐ x ∂μ, (∑' i, nnnorm (f (i + 1) x - f i x) : ℝ≥0∞) < ∞ :=
+begin
+  have hp_pos : 0 < p := zero_lt_one.trans_le hp1,
+  have h_integral : ∫⁻ a, (∑' i, nnnorm (f (i + 1) a - f i a) : ℝ≥0∞)^p ∂μ < ∞,
+  { have h_tsum_lt_top : (∑' i, B i) ^ p < ∞,
+      from ennreal.rpow_lt_top_of_nonneg hp_pos.le (lt_top_iff_ne_top.mp hB),
+    refine lt_of_le_of_lt _ h_tsum_lt_top,
+    rwa [←@ennreal.le_rpow_one_div_iff _ _ (1/p) (by simp [hp_pos]), one_div_one_div] at h, },
+  have rpow_ae_lt_top : ∀ᵐ x ∂μ, (∑' i, nnnorm (f (i + 1) x - f i x) : ℝ≥0∞)^p < ∞,
+  { refine ae_lt_top' (ae_measurable.ennreal_rpow_const _) h_integral,
+    exact ae_measurable.ennreal_tsum (λ n, ((hf (n+1)).sub (hf n)).nnnorm.ennreal_coe), },
+  refine rpow_ae_lt_top.mono (λ x hx, _),
+  rwa [←ennreal.lt_rpow_one_div_iff hp_pos,
+    @ennreal.top_rpow_of_pos (1/p) (by simp [hp_pos])] at hx,
+end
+
+lemma ae_tendsto_of_cauchy_snorm' [complete_space E] {f : ℕ → α → E} {p : ℝ}
+  (hf : ∀ n, ae_measurable (f n) μ) (hp1 : 1 ≤ p) {B : ℕ → ℝ≥0∞} (hB : ∑' i, B i < ∞)
+  (h_cau : ∀ (N n m : ℕ), N ≤ n → N ≤ m → snorm' (f n - f m) p μ < B N) :
+  ∀ᵐ x ∂μ, ∃ l : E, filter.at_top.tendsto (λ n, f n x) (𝓝 l) :=
```

Comment:
I wonder if, once this lemma is proved (as well as its analogue for infinite p), you can not already reunite the finite `p` and infinite `p` cases using a general reasoning. I mean the following:
* First, prove a general statement that if `f n` tends pointwise to `f` and all `f n` are ae-measurable (where `n` can live in any index set), then `snorm f p μ \le liminf snorm (f n) p μ`. This is true for all `p`, and this is a useful general lemma, so it is good to have it in any case. The proof of the lemma will distinguish between finite and infinite p, of course, but the statement is general.
* From there, once you have almost sure convergence both for finite p and infinite p, then I think you can conclude without ever having to distinguish between the two.

This would have the advantage of extracting a useful lemma from your proof, and avoiding some amount of code duplication. What do you think?